txpipe/theory.py

```diff
@@ -103,81 +110,85 @@ def replace_with_theory_real(self, s, cosmo):
         ell = np.unique(np.logspace(np.log10(2), 5, 400).astype(int))
         tracers = self.get_ccl_tracers(s, cosmo)
 
-        for i in range(nbin_source):
-            for j in range(i + 1):
-                print(f"Computing theory lensing-lensing ({i},{j})")
+        if "galaxy_shear_xi_plus" and "galaxy_shear_xi_minus" in s.get_data_types():
```

Comment:
This line isn't doing what you think it's doing. `"galaxy_shear_xi_plus" and "galaxy_shear_xi_minus"` will always evaluate to `"galaxy_shear_xi_plus"`, for weird python reasons. You need `"galaxy_shear_xi_plus" in s.get_data_types() and "galaxy_shear_xi_minus" in s.get_data_types()` or equivalent.  Or you could just assume that if xip exists then so does xim.

Reply:
oh I see, interesting! Fixed.